sql/plan/delete.go

```diff
@@ -15,23 +15,61 @@
 package plan
 
 import (
+	"fmt"
+	"strings"
+
 	"gopkg.in/src-d/go-errors.v1"
 
 	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/transform"
 )
 
 var ErrDeleteFromNotSupported = errors.NewKind("table doesn't support DELETE FROM")
 
 // DeleteFrom is a node describing a deletion from some table.
 type DeleteFrom struct {
 	UnaryNode
+	// targets are the explicitly specified table nodes from which rows should be deleted. For simple DELETES against a
+	// single source table, targets do NOT need to be explicitly specified and will not be set here. For DELETE FROM JOIN
+	// statements, targets MUST be explicitly specified by the user and will be populated here.
```

Comment:
so there are three input formats? (1) implicit DELETE, (2) DELETE with one target, and (3) DELETE with two targets? or are 1/2 collapsed by parser?

Reply:
reading the comments below it seems like it's either/or. it almost might make more sense to make a DeleteMulti node if the behavior is so different? the hard part then would be finding everywhere we switch on Delete and Insert nodes, and adding the new node.

Reply:
@zachmu might have a clever idea of how to bridge or whether to separate the two types of deletes

Reply:
Chatted in person about some other ways to set up the plan that might simplify the delete from join logic. Max suggested a `broadcast/spread` and `coalesce` pattern for the node tree that seems like an elegant way to simplify the logic by restructuring the plan, but there are some issues with how transforms would modify the tree and cause the `DeleteFrom` nodes to stop sharing the same child `broadcast` node instance. Memo groups could potentially be one way to solve that in the future.